### PR TITLE
Add Drift Detection Status Fields 

### LINKED ISSUE #7: Add field in status when applied by drift detection

### PR DESCRIPTION
**Pull request by phoban01**
This PR adds the following status fields to the Terraform resource:
-  `LastDriftDetectedAt`
-  `LastAppliedByDriftDetectionAt`

Also includes a fix for a sneaky variable shadowing problem in the drift detection code.

Fixes #7

### CODE REVIEW

**Comment by chanwit** [comment]
LGTM @phoban01 

could we wait until @tomhuang12 Health Check gets merged please.
His #53 PR is rather big and it would be harder to rebase than this one.

**Comment by phoban01** [comment]
@chanwit I think this is good to go now.

**Review by chanwit — APPROVED** [review]
LGTM thank you @phoban01 :1st_place_medal: !!

**Pull request merged**
